update version and CHANGELOG.md (#591)
ROCm/rocWMMA: CHANGELOG.md

Documentation for rocWMMA is available at
[https://rocm.docs.amd.com/projects/rocWMMA/en/latest](https://rocm.docs.amd.com/projects/rocWMMA/en/latest).

## (Unreleased) rocWMMA 2.1.0 for ROCm 7.1.0

### Added

* Added more unit tests to increase the code coverage.
* Added sample `perf_i8gemm` to demonstrate `int8_t` as matrix input data type.

### Changed

* Increased compile timeout and improved visualization in `math-ci`.

### Removed

> * Removed absolute paths from the RPATH of sample and test binary files.

Reviewer: ```suggestion
* Removed absolute paths from RPATH and test binaries.
```
Could we phrase it more simply as above?

Reviewer 2: I believe the original wording is more accurate.

Author: sample and test are two distinct types of binaries in our library. Thanks for the suggestion, though.

> * Removed absolute paths from the RPATH of sample and test binary files.

Reviewer 2: ```suggestion
* Removed absolute paths from the `RPATH` of sample and test binary files.
```

Author: changed it accordingly, thanks.

### Resolved issues

> * Fixed issues caused by HIP changes: removed the .data member from HIP_vector_type, and broadcast constructor now only writes to the first vector element.

Reviewer: ```suggestion
* Fixed issues caused by HIP changes: 
   * Removed the `.data` member from `HIP_vector_type`.
   * Broadcast constructor now only writes to the first vector element.
```
Consider breaking this into sub-bullet points for better readability.

Reviewer 2: If you add the sub-bullet points, please also include the next item:
```
   * Fixed a bug related to `int32_t` usage in `hipRTC_gemm` for gfx942.
```

Author: Changed accordingly, thanks

* Fixed a bug related to `int32_t` usage in `hipRTC_gemm` for gfx942, caused by breaking changes in HIP.
* Replaced `#pragma unroll` with `static for` to fix a bug caused by the upgraded compiler which no longer supports using `#pragma unroll` with template parameter indices.
* Corrected test predicates for `BLK` and `VW` cooperative kernels.

> * Modified `compute_utils.sh` in `build-infra` to ensure rocWMMA is built with gfx1151 target for ROCm 7.0 and beyond.

Reviewer 2: Is this change in the rocWMMA repo?

Author: The "compute_utils.sh" is not in rocWMMA repo. It's in this repo: https://github.com/AMD-ROCm-Internal/build-infra. You need your _amdeng user name to access it. All CI pipelines rely on it to build the libraries.
The related change is in this PR: https://github.com/AMD-ROCm-Internal/build-infra/pull/178.

Reviewer 2: Thanks Ryker!
